### DISCUSSION TITLE
GUI improvements for usability with the keyboard.

### DISCUSSION THREAD

**Discussion by AverageHoarder** [qarmin/czkawka#1103] [Ideas]
Hi, I've used czkawka for the first time today and am impressed by the performance.
However there are a few things that would be lovely features/improvements in my opinion.

### Autoscroll to keep selection within visible part of the GUI
I like to quickly go over lists of duplicates via the keyboard, using the up and down arrow keys.
In Czkawka when I reach the bottom of the screen the selection disappears behind the bottom text panel (also happens when that panel is minimized) and only starts scrolling down when I press the down arrow key a few more times.
In those cases the selection is not visible on screen and I have to scroll down with my mouse to once again see what I'm doing.

### Allow skipping of the blank lines between duplicates when using a keyboard to navigate
While I like that there are blank lines between sets of detected duplicates to separate them visually I'd like to be able to skip them when navigating file lists via keyboard (arrow keys).
I don't know a reason why I would want to select a blank line via keyboard and it's an extra keystroke that's avoidable in my opinion.

### Allow custom hotkey for selecting files
Maybe I've missed it but I didn't see a mention in the documentation as to how to select files via the keyboard.
I found out via trial and error that the default key to select or deselect a file is "a". However I'm very used from other programs to select stuff via the spacebar and other people might prefer other binds.
While I can get used to using "a" to select I'd prefer the spacebar or the ability to choose which key to use.

Otherwise I'm really happy with Czkawka so far. Beats the crap out of ancient and bloatware infested programs I used to use 10 years ago.
Btw. if it affects the problems I have described, I'm using the windows gui+cli version of Czkawka 6.1.0.

**Comment by Alister-Zeus** [comment]
I found if I click and select with <rtn> then I can move <up> & <down> until the window loses focus then I have to reclick before <rtn> works again

**Comment by EvocativeOlash** [comment]
Remains a problem, especially with dark theme the selection is low contrast and on large lists on similar files its very fatiguing when you have to constantly search for where the selection cursor has gone off screen.
